### DISCUSSION TITLE
One-way relations

### DISCUSSION THREAD

**Discussion by shinypb** [prisma/prisma#10360] [Q&A] [answered]
Hello, world! I'm new to Prisma (and, honestly, ORMs in general), and I could use a bit of advice on the best way to accomplish something.

First, let me lay out what I think I want and how I thought I'd be able to accomplish it, and then please tell me the best Prisma-y practices way you'd approach it. 😅  Feedback and guidance of all kinds are very welcome and appreciated.

So: what I _want _is a one-way relation, but that doesn't seem to be something that Prisma (currently supports). Imagine a product Twitch, where there are **users** who have zero or more **broadcasts**. At any given time, they may or may not be actively broadcasting.

What I'd like to see is a `broadcast` property on the `User` model that is either null (meaning: not broadcasting) or points to whichever `Broadcast` is currently happening. I'd like my `Broadcast` model to have a relation back to the user that broadcasted it.

I've got two models that have a relation:

````prisma
model User {
    id           String     @id @default(cuid())
    broadcastId  String?
    broadcast    Broadcast?
}

model Broadcast {
    id           String  @id @default(cuid())
    userId       String
    user         User    @relation(fields: [userId], references: [id])
}
````

This schema works fine if there's either zero or one broadcast associated with a particular user, but fails if I try to add a second broadcast because relations are always unique.

So, my question to the room is: what's the _correct_ way to model this instead?

**Comment by austincrim** [comment, marked as answer]
Hey @shinypb 👋, welcome to the Prisma community! We are excited to have you.

If a `User` is expected to have more than one `Broadcast`, you will need to model it as a 1-to-many relationship like this:
```prisma
model User {
  id        String      @id @default(cuid())
  broadcast Broadcast[]
}

model Broadcast {
  id       String  @id @default(cuid())
  user     User    @relation(fields: [userId], references: [id])
  userId   String
  isActive Boolean
}
```

You will also notice I added an `isActive` boolean on the `Broadcast`, which could be a way to to determine the currently active `Broadcast`.

You can read more about one-to-many relationships [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/one-to-many-relations)

Let me know if you have more questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
